### PR TITLE
NetworkWatcher : Adding VMSS as a source in ConnectionMonitor

### PR DESCRIPTION
**Pull request by kumaam**
## Description

NetworkWatcher : Adding VMSS as a source in ConnectionMonitor.

Notes : 
* Added new Testcase for NetworkWatcher - ConnectionMonitor with VMSS as a source.
* Updated ConnectionMonitor wiki file.

Testing Details
Ran below commands to test the changes and connection monitor gets created successfully
```
$SourceEndpointVMSSObject = New-AzNetworkWatcherConnectionMonitorEndpointObject -Name "vmssEndpoint" -AzureVMSS -ResourceId "/subscriptions/9cece3e3-0f7d-47ca-af0e-9772773f90b7/resourceGroups/Piyush-OMS-Test/providers/Microsoft.Compute/virtualMachineScaleSets/vmsscuseuap" 

$testGroup = New-AzNetworkWatcherConnectionMonitorTestGroupObject -Name testGroupName -TestConfiguration $testConfiguration -Source $SourceEndpointVMSSObject -Destination $destinationEndpoint

$cm = New-AzNetworkWatcherConnectionMonitor -NetworkWatcherName $networkWatcherName -ResourceGroupName "NetworkWatcherRG" -Name $testname -TestGroup $testGroup
```

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by wyunchi-ms** [comment]
Please update the changelog.

File: src/Network/Network/NetworkWatcher/ConnectionMonitor/NewAzureNetworkWatcherConnectionMonitorEndPointObjectCommand.cs [hunk 1]
```diff
@@ -131,6 +145,10 @@ public class NewAzureNetworkWatcherConnectionMonitorEndpointObjectCommand : Conn
             Mandatory = false,
             HelpMessage = "List of items which need to be excluded from endpoint scope.",
              ParameterSetName = "MMAWorkspaceNetwork")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "List of items which need to be excluded into endpont scope.",
```
Comment (kumaam) [inline src/Network/Network/NetworkWatcher/ConnectionMonitor/NewAzureNetworkWatcherConnectionMonitorEndPointObjectCommand.cs:150]: Spelling

File: src/Network/Network.Test/ScenarioTests/NetworkWatcherAPITests.cs [hunk 2]
```diff
@@ -114,6 +114,14 @@ public void TestConnectionMonitor()
             TestRunner.RunTestScript("Test-ConnectionMonitor");
         }
 
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        [Trait(Category.Owner, NrpTeamAlias.netanalyticsdev)]
+        public void TestConnectionMonitorVMSS()
+        {
+            TestRunner.RunTestScript("Test-ConnectionMonitorVMSS");
```
Comment (kumaam) [inline src/Network/Network.Test/ScenarioTests/NetworkWatcherAPITests.cs:122]: Rename - Test-ConnectionMonitorVMSSAsSource

File: src/Network/Network.Test/ScenarioTests/NetworkWatcherAPITests.ps1 [hunk 3]
```diff
@@ -1394,4 +1437,83 @@ function Test-ConnectionMonitor
...
+
+        #Get Vmss and Instances
+        $vmss = Get-AzVmss -ResourceGroupName $resourceGroupName -VMScaleSetName $virtualMachineScaleSetName
+
+        #Install networkWatcherAgent on Vmss and Vmss Instances
+        Add-AzVmssExtension -VirtualMachineScaleSet $vmss -Name "AzureNetworkWatcherExtension" -Publisher "Microsoft.Azure.NetworkWatcher" -Type "NetworkWatcherAgentWindows" -TypeHandlerVersion "1.4" -AutoUpgradeMinorVersion $True
+        Update-AzVmss -ResourceGroupName "$resourceGroupName" -Name $virtualMachineScaleSetName -VirtualMachineScaleSet $vmss
+
+        $instances = Get-AzVmssVM -ResourceGroupName "$resourceGroupName" -VMScaleSetName $vmss.Name
+        foreach($item in $instances) {
```
Comment (kumaam) [inline src/Network/Network.Test/ScenarioTests/NetworkWatcherAPITests.ps1:1487]: Add a comment why we need to manually update each VMs in VMSS.

**Comment by kumaam** [comment]
> Please update the changelog.

Updated the changelog

**Comment by kumaam** [comment]
Review is pending

**Comment by wyunchi-ms** [comment]
LGTM, but is the target branch `network-may-2022` or `release-2022-05-24`?

**Comment by kumaam** [comment]
> LGTM, but is the target branch `network-may-2022` or `release-2022-05-24`?

Hi, This change has a dependency on Swagger changes which is not merged/deployed yet. Hence, we will wait to merge it in release branch (that's why target branch is network-may-2022).

**Comment by wyunchi-ms** [comment]
> > LGTM, but is the target branch `network-may-2022` or `release-2022-05-24`?
> 
> Hi, This change has a dependency on Swagger changes which is not merged/deployed yet. Hence, we will wait to merge it in release branch (that's why target branch is network-may-2022).

Please keep the tag `Do Not Merge` until your swagger is ready and ping us.

**Comment by kumaam** [comment]
Moved to this PR: https://github.com/Azure/azure-powershell/pull/19002

**Pull request closed without merging**